servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -723,17 +749,24 @@ void resetError() {
 
     static final class RetryingServiceDiscoverer<U, R, E extends ServiceDiscovererEvent<R>>
             extends DelegatingServiceDiscoverer<U, R, E> {
-        private final ServiceDiscoveryRetryStrategy<R, E> retryStrategy;
+        @Nullable
+        private final BiIntFunction<Throwable, ? extends Completable> retryStrategy;
+        private final ServiceDiscoveryRetryStrategy<R, E> deprecatedRetryStrategy;
 
         RetryingServiceDiscoverer(final ServiceDiscoverer<U, R, E> delegate,
-                                  final ServiceDiscoveryRetryStrategy<R, E> retryStrategy) {
+                                  @Nullable final BiIntFunction<Throwable, ? extends Completable> retryStrategy,
+                                  final ServiceDiscoveryRetryStrategy<R, E> deprecatedRetryStrategy) {
             super(delegate);
-            this.retryStrategy = requireNonNull(retryStrategy);
+            this.retryStrategy = retryStrategy;
+            this.deprecatedRetryStrategy = requireNonNull(deprecatedRetryStrategy);
```

Comment:
Consider asserting that only one of the retry strategies is not null

Reply:
I added validation in the constructor.